CMakeLists.txt

```diff
@@ -125,15 +125,67 @@ if("sgx" IN_LIST COMPILE_TARGETS)
   )
 
   add_dependencies(ccf ccf.enclave)
-endif()
 
-if("virtual" IN_LIST COMPILE_TARGETS)
+elseif(COMPILE_TARGET STREQUAL "snp")
+
+  # virtual version
+  add_library(ccf.virtual STATIC ${CCF_IMPL_SOURCE})
```

Comment:
Maybe not in this PR, but we want to produce distinct artefacts, very likely with a different toolchain and flags.

Reply:
Agreed, as per [Juliens suggestion](https://github.com/microsoft/CCF/pull/4367#pullrequestreview-1154913030) I've added a comment to that effect